src/main/java/seedu/address/model/datetime/DatetimeCommonUtils.java

```diff
@@ -68,6 +77,72 @@ public static String dayOfWeekToReadable(DayOfWeek day) {
         return day.getDisplayName(TextStyle.SHORT, Locale.getDefault());
     }
 
+    /**
+     * Splits a time range string by the hyphen "-".
+     * Checks whether the start time and end time is logically valid
+     *
+     * @param timeRangeString  String to be split
+     * @throws ParseException If the string does not appear to be a time range
+     */
+    public static String[] splitTimeRangeString(String timeRangeString) throws ParseException {
+        if (!timeRangeString.matches(TIMERANGE_FORMAT_REGEX)) {
+            throw new ParseException(TIMERANGE_MESSAGE_CONSTRAINTS);
+        }
+        return timeRangeString.trim().split("-");
+    }
+
+    /**
+     * Checks whether the date string is valid.
+     * The strings must be parsable by LocalTime.
+     *
+     * @param dateString Date string
+     * @throws ParseException If the date is invalid.
+     */
+    public static void assertDateValid(String dateString) throws ParseException {
+        try {
+            LocalTime.parse(dateString, DATE_INPUT_FORMATTER);
```

Comment:
It should be LocalDate

Reply:
Also need to add PREFIX_DAY to add consultation command parser since we change the format of this command.

Reply:
Also, in `AddConsultationCommandParser`
` ParserUtil.assertAllPrefixesPresent(...)` missing `PREFIX_DAY`

Reply:
Since we add 1 more parameter (DAY) to consultation, we need to change `ConsultationBuilder` in `test` also.